### PR TITLE
update doc to comply with HCLv2 and Ignition v2

### PR DESCRIPTION
**Pull request by egeturgay**
Reflects the changes on Ignition v2 and HCL v2 as explained in https://github.com/community-terraform-providers/terraform-provider-ignition/issues/4

### CODE REVIEW

**Review by yussufsh — APPROVED** [review]
/lgtm

File: website/docs/d/directory.html.md [hunk 1]
```diff
@@ -23,11 +22,11 @@ data "ignition_directory" "folder" {
 
 The following arguments are supported:
 
-* `filesystem` - (Required) The internal identifier of the filesystem. This matches the last filesystem with the given identifier. This should be a valid name from a _ignition\_filesystem_ resource.
-
 * `path` - (Required) The absolute path to the directory.
 
-* `mode` - (Optional) The directory's permission mode. Note that the mode can be specified as either an octal value (e.g 0755) or a decimal value (i.e. 493 as equivalent to the octal 0755).
+* `overwrite` - (Optional) Whether to delete preexisting nodes at the path
```
Comment (LorbusChris) [inline website/docs/d/directory.html.md]: it would be great to menteion the default here: `Defaults to false.`

File: website/docs/d/file.html.md [hunk 2]
```diff
@@ -42,17 +40,17 @@ data "ignition_file" "hello" {
 
 The following arguments are supported:
 
-* `filesystem` - (Required) The internal identifier of the filesystem. This matches the last filesystem with the given identifier. This should be a valid name from a _ignition\_filesystem_ resource.
-
 * `path` - (Required) The absolute path to the file.
 
+* `overwrite` - (Optional) Whether to delete preexisting nodes at the path. Defaults to true.
```
Comment (LorbusChris) [inline website/docs/d/file.html.md]: Please double check. I believe in Ignition v2 / spec v3.x, this defaults to `false`. (used to be `true` in v0.x / spec 2.x)

File: website/docs/d/file.html.md [hunk 3]
```diff
@@ -42,17 +40,17 @@ data "ignition_file" "hello" {
...
+
 * `content` - (Optional) Block to provide the file content inline.
 
 * `source` - (Optional) Block to retrieve the file content from a remote location.
 
 	__Note__: `content` and `source` are mutually exclusive.
 
-* `mode` - (Optional) The file's permission mode. Note that the mode can be specified as either an octal value (e.g 0755) or a decimal value (i.e. 493 as equivalent to the octal 0755).
+* `mode` - (Optional) The file's permission mode. Note that the mode must be properly specified as a decimal value (i.e. 0755 -> 493).
```
Comment (LorbusChris) [inline website/docs/d/file.html.md]:
```suggestion
* `mode` - (Optional) The file's permission mode. Note that the mode must be properly specified as a decimal value, not octal (i.e. 0755 -> 493).
```

File: website/docs/d/directory.html.md [hunk 4]
```diff
@@ -23,11 +22,11 @@ data "ignition_directory" "folder" {
...
+
+* `mode` - (Optional) The directory's permission mode. Note that the mode must be properly specified as a decimal value (i.e. 0755 -> 493).
```
Comment (LorbusChris) [inline website/docs/d/directory.html.md]:
```suggestion
* `mode` - (Optional) The directory's permission mode. Note that the mode must be properly specified as a decimal value, not octal (i.e. 0755 -> 493).
```

**Review by LorbusChris — CHANGES_REQUESTED** [review]
@egeturgay my apologies for the wait, this totally slipped by me. Thank you for submitting this!

**Comment by LorbusChris** [comment]
@egeturgay I left some comments. Could you have another look?

File: website/docs/d/directory.html.md (hunk 1 above)
Comment (egeturgay) [reply website/docs/d/directory.html.md]: will add in the next commit

File: website/docs/d/file.html.md (hunk 2 above)
Comment (egeturgay) [reply website/docs/d/file.html.md]: your suggestion seems correct according to https://github.com/coreos/ignition/blob/master/docs/configuration-v3_2.md
I've updated it

**Comment by egeturgay** [comment]
thanks @LorbusChris! I've updated it with the requested changes.

**Review by LorbusChris — APPROVED** [review]

**Pull request merged**